mdata/aggmetrics.go

```diff
@@ -49,8 +44,8 @@ func (ms *AggMetrics) GC() {
 		time.Sleep(diff + time.Minute)
 		log.Info("checking for stale chunks that need persisting.")
 		now := uint32(time.Now().Unix())
-		chunkMinTs := now - (now % ms.chunkSpan) - uint32(ms.chunkMaxStale)
-		metricMinTs := now - (now % ms.chunkSpan) - uint32(ms.metricMaxStale)
+		chunkMinTs := now - uint32(ms.chunkMaxStale)
+		metricMinTs := now - uint32(ms.metricMaxStale)
```

Comment:
The old code calculated the last chunk T0 that is older then chunkMaxStale.   I believe this is because the logic in AggMetric.GC() compared hte chunkMinTs and metricMinTs to the chunk.T0,  but as we now compare against chunk.LastWrite this new way should be fine.

The question this raises though, is should our maxStale settings be per retention policy? If a MT service is receiving some metrics every 10seconds and some every few hours, then applying the same maxStale settings to both is not ideal.